prediction_market_agent/agents/microchain_agent/microchain_agent.py

```diff
@@ -108,12 +123,21 @@ def main(
         bootstrap=TRADING_AGENT_BOOTSTRAP,  # Same here.
         long_term_memory=long_term_memory,
         allow_stop=False,  # Prevent the agent from stopping itself
+        prompt_handler=prompt_handler if load_historical_prompt else None,
     )
     if seed_prompt:
         agent.bootstrap = [f'Reasoning("{seed_prompt}")']
     agent.run(iterations=iterations)
     # generator.print_usage() # Waiting for microchain release
     long_term_memory.save_history(agent.history)
+    editable_prompt = get_editable_prompt_from_agent(agent)
+    prompt_handler.save_prompt(editable_prompt)
+
+
+def get_editable_prompt_from_agent(agent: Agent) -> str:
+    # Split prompt into editable, non-editable
+    editable_prompt = str(agent.system_prompt).split(NON_UPDATABLE_DIVIDOR)[0].strip()
```

Comment:
```suggestion
    editable_prompt =agent.system_prompt.split(NON_UPDATABLE_DIVIDOR)[0].strip()
```

Reply:
Closing this one as microchain's system prompt isn't typed, so we need this weird str(...) :(